WatchApp Extension/Controllers/ActionHUDController.swift

```diff
@@ -81,6 +81,28 @@ final class ActionHUDController: HUDInterfaceController {
 
     // MARK: - Menu Items
 
+    private func setWorkoutMode(duration: Double) {
+        guard var glucoseTargetRangeSchedule = loopManager.settings.glucoseTargetRangeSchedule else {
+            return
+        }
+        let endDate = Date().addingTimeInterval(TimeInterval(hours: duration))
+        guard glucoseTargetRangeSchedule.setOverride(.workout, until: endDate) else {
+            return
+        }
+        sendGlucoseRangeSchedule(glucoseTargetRangeSchedule)
+    }
+    
+    // Add force-touch menu items to set workout mode for certain fixed durations:
+    @IBAction func setWorkout30Min() {
```

Comment:
Should use a a long-press gesture on the workout mode button to provide options for workout mode duration, that would present an action sheet (WKAlertControllerStyleActionSheet) with the time options.

Reply:
Long-press (force touch) is sort of a one-trick pony in terms of interface.   It can only apply to the whole screen (can't be associated with a button), and the only thing it can do is to present a context menu with simple buttons.  So I think there are two choices:  keep the code as-is (with the admittedly odd combinations of two different ways to set workout mode), or change the function of the Workout button to present the action sheet, including both the finite time and indefinite time options.  The latter is definitely more consistent with the main Loop app interface.  The only thing it loses is the ability to set Workout mode indefinitely with one press - it would now require two presses.  That's probably a worthwhile tradeoff for a cleaner interface.  What do you think?

Reply:
Ah, ok!  Thanks @mpangburn .  I'll take a look.